website/server/controllers/api-v3/tasks.js

```diff
@@ -838,6 +838,28 @@ api.moveTask = {
 
     // In memory updates
     const order = owner.tasksOrder[`${task.type}s`];
+    if (order.indexOf(task._id) === -1) { // task is missing from list, list needs repair
+      const taskListQuery = { type: task.type };
+      if (group) {
+        taskListQuery['group.id'] = owner._id;
+      } else if (challenge) {
+        taskListQuery['challenge.id'] = owner._id;
```

Comment:
I believe this search will return both the Challenge's own tasks and copies of the tasks that are in participants' task lists. It would need to also check for userId not existing, to get only the former.

From memory, something similar may happen for the group search too.